source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveContainerRenderer.cs

```diff
@@ -193,18 +258,28 @@ public static bool ApplyPadding(Border border, Grid uiElement, AdaptiveCollectio
 
                 if (element.Bleed)
                 {
-                    int leftMargin = 0, rightMargin = 0;
-                    if (parentRenderArgs.BleedDirection == BleedDirection.Left || parentRenderArgs.BleedDirection == BleedDirection.Both)
+                    int leftMargin = 0, rightMargin = 0, topMargin = 0, bottomMargin = 0;
+                    if ((parentRenderArgs.BleedDirection & BleedDirection.BleedLeft) != BleedDirection.BleedNone)
                     {
                         leftMargin = -padding;
                     }
 
-                    if (parentRenderArgs.BleedDirection == BleedDirection.Right || parentRenderArgs.BleedDirection == BleedDirection.Both)
+                    if ((parentRenderArgs.BleedDirection & BleedDirection.BleedRight) != BleedDirection.BleedNone)
                     {
                         rightMargin = -padding;
                     }
 
-                    border.Margin = new Thickness { Left = leftMargin, Right = rightMargin };
+                    if ((parentRenderArgs.BleedDirection & BleedDirection.BleedUp) != BleedDirection.BleedNone)
+                    {
+                        topMargin = -padding;
+                    }
+
+                    if ((parentRenderArgs.BleedDirection & BleedDirection.BleedDown) != BleedDirection.BleedNone)
+                    {
+                        bottomMargin = -padding;
+                    }
+
+                    border.Margin = new Thickness { Left = leftMargin, Top = topMargin, Right = rightMargin, Bottom = bottomMargin };
```

Comment:
This feels like it should be in a helper... Perhaps something like

```
Margin GetBleedMargin(RenderArgs parentRenderArgs, int padding) {}
```
?

Reply:
Or maybe even just as a member function hanging off of `AdaptiveRenderArgs`?

---
In reply to: [283556686](https://github.com/microsoft/AdaptiveCards/pull/2879#discussion_r283556686) [](ancestors = 283556686)